kitsune/users/auth.py

```diff
@@ -195,30 +197,36 @@ def update_user(self, user, claims):
         # Check if the user has active subscriptions
         subscriptions = claims.get("subscriptions", [])
 
+        # The request attribute might not be set.
+        request = getattr(self, "request", None)
+
         if not profile.is_fxa_migrated:
```

Comment:
You can move the check for the existence of the request object here. This error is triggered from events that are coming from FxA thus it makes sense to assume that the profiles are migrated already. This way you can avoid the 2 checks under this if statement

```python
if (request := getattr(self, "request", None)) and not profile.is_fxa_migrated:
```

Reply:
Another way to go about this would be to check for the absence of the request, do the relevant tasks and then return.

```python
if not request:
   we_are_doing_all_the_common_tasks_for_both_the_login_flow_and_the_celery_task()
   return
we_are_now_only_in_the_login_flow_and_we_have_the_request_object_steps